pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -13,6 +13,65 @@ import (
 	"github.com/holiman/uint256"
 )
 
+
+// getHighLen calculates the highest bit length of `scalar_u.d2` and `scalar_v.d2`,
+// subtracts 1 from the result, and assigns it to `ids.len_hi`. 
+//
+// `newGetHighLenHint` takes three operanders as arguments:
+//   - `len_hi`: the variable that will store the result of the bit-length calculation.
+//   - `scalar_u.d2`: the first scalar value.
+//   - `scalar_v.d2`: the second scalar value.
+//
+// The function resolves these operanders, calculates the maximum bit length of `scalar_u.d2` and `scalar_v.d2`,
+// and assigns the result (bit_length - 1) to `len_hi`.
+func newGetHighLenHint(len_hi, scalar_u, scalar_v hinter.ResOperander) hinter.Hinter {
+    return &hinter.GenericZeroHinter{
+        Name: "GetHighLen",
+        Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+            //> ids.len_hi = max(ids.scalar_u.d2.bit_length(), ids.scalar_v.d2.bit_length())-1
+
+            scalarU, err := hinter.ResolveAsFelt(vm, scalar_u)
```

Comment:
if you look at this [link](https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/highest_bitlen.cairo#L9), you will notice, that `scalar_u` and `scalar_v` are actually of type BigInt3. In the `memory` package we have a function to resolve the reference as BigInt3, you can use it like this: `scalarUValues, err := vm.Memory.ResolveAsBigInt3(scalar_u)`. It returns an array of 3 felts, where d2 is the third I suppose. You can use the `bit_length()` method on it, and leave the rest of the implementation as it is, because it looks solid.

Reply:
Reference: https://github.com/NethermindEth/cairo-vm-go/blob/a55dae8d65d234b6e5a6b2d53cfbfb3d096c377d/pkg/hintrunner/zero/zerohint_ec.go#L37

Reply:
I appreciate the feedback and guidance, tomorrow I will apply these changes asap thanks 🫡